ui/src/app/components/ct-available-banner-maybe.tsx

```diff
@@ -0,0 +1,69 @@
+import * as React from 'react';
+import {useEffect, useState} from 'react';
+
+import {ToastBanner, ToastType} from './toast-banner';
+import {cookiesEnabled} from 'app/utils/cookies';
+import {StyledRouterLink} from './buttons';
+import {environment} from 'environments/environment';
+import {AccessTierShortNames} from 'app/utils/access-tiers';
+import {CdrVersionTier, Profile} from 'generated/fetch';
+import {eligibleForTier} from 'app/utils/access-utils';
+import {cdrVersionStore, profileStore, useStore} from 'app/utils/stores';
+
+// Visible For Testing
+export const DAR_PATH = '/data-access-requirements';
+const CT_COOKIE_KEY = 'controlled-tier-available';
+
+const shouldShowBanner = (profile: Profile, cdrVersionTiers: CdrVersionTier[]) => {
+  const ct = cdrVersionTiers?.find(v => v.accessTierShortName === AccessTierShortNames.Controlled);
+
+  // all of the following must be true
+  const shouldShow = profile && ct &&
+    // the environment allows users to see the CT (in the UI)
+    environment.accessTiersVisibleToUsers.includes(AccessTierShortNames.Controlled) &&
+    // the user is eligible for the CT
+    eligibleForTier(profile, AccessTierShortNames.Controlled) &&
+    // the user does not currently have CT access
+    !profile.accessTierShortNames.includes(AccessTierShortNames.Controlled) &&
```

Comment:
How is  eligibleForTier(profile, AccessTierShortNames.Controlled) and !profile.accessTierShortNames.includes(AccessTierShortNames.Controlled) different?

Reply:
nvm